amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/polling/KinesisDataFetcher.java

```diff
@@ -120,7 +121,13 @@ public DataFetcherResult getRecords() {
 
         if (nextIterator != null) {
             try {
-                return new AdvancingResult(getRecords(nextIterator));
+                GetRecordsResponse getRecordsResponse = getRecords(nextIterator);
+                while (!isValidResponse(getRecordsResponse)) {
```

Comment:
I think this during a service-side event this might be problematic. If we start having many retries we can start getting throttled which will cause a loop where we keep getting invalid responses. We should add some retry strategy with backoff, or not bother retrying and just propagate some exception.